### PR TITLE
chore: Upgrade constructs

### PR DESCRIPTION
**Pull request by iliapolo**
Upgrade construct to version `3.0.4`. This version does not include the tokens mechanism, to replace it, we implemented a simple mechanism that facilitates just the cdk8s needs for tokens:

- Lazy.any
- Implicit token generated by the CLI (objects with a `resolve` function)

This PR enabled https://github.com/aws/aws-cdk/issues/9670

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: examples/typescript/hello/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -12,7 +12,7 @@ Array [
       "ports": Array [
         Object {
           "port": 80,
-          "targetPort": 8080,
```
Comment (iliapolo) [inline examples/typescript/hello/__snapshots__/index.test.js.snap:15]: Check this

File: package.json [hunk 2]
```diff
@@ -24,7 +25,9 @@
       "cdk8s/follow-redirects",
       "cdk8s/follow-redirects/**",
       "cdk8s-plus/minimatch",
-      "cdk8s-plus/minimatch/**"
+      "cdk8s-plus/minimatch/**",
+      "projen/constructs",
```
Comment (iliapolo) [inline package.json:29]: This isn't needed anymore - projen doesn't use constructs.

File: packages/cdk8s-plus/src/duration.ts [hunk 3]
```diff
@@ -92,7 +89,7 @@ export class Duration {
   private readonly unit: TimeUnit;
 
   private constructor(amount: number, unit: TimeUnit) {
-    if (!Token.isUnresolved(amount) && amount < 0) {
```
Comment (iliapolo) [inline packages/cdk8s-plus/src/duration.ts:95]: Check this

File: packages/cdk8s-plus/src/duration.ts [hunk 4]
```diff
@@ -209,20 +205,6 @@ export class Duration {
     }
   }
 
-  /**
-   * Returns a string representation of this `Duration` that is also a Token that cannot be successfully resolved. This
-   * protects users against inadvertently stringifying a `Duration` object, when they should have called one of the
-   * `to*` methods instead.
-   */
-  public toString(): string {
```
Comment (iliapolo) [inline packages/cdk8s-plus/src/duration.ts:217]: Check this

File: packages/cdk8s-plus/src/duration.ts [hunk 5]
```diff
@@ -270,9 +252,6 @@ function convert(amount: number, fromUnit: TimeUnit, toUnit: TimeUnit, { integra
   if (fromUnit.inMillis === toUnit.inMillis) { return amount; }
   const multiplier = fromUnit.inMillis / toUnit.inMillis;
 
-  if (Token.isUnresolved(amount)) {
```
Comment (iliapolo) [inline packages/cdk8s-plus/src/duration.ts:273]: Check this

File: packages/cdk8s-plus/src/size.ts [hunk 6]
```diff
@@ -53,7 +51,7 @@ export class Size {
   private readonly unit: StorageUnit;
 
   private constructor(amount: number, unit: StorageUnit) {
-    if (!Token.isUnresolved(amount) && amount < 0) {
```
Comment (iliapolo) [inline packages/cdk8s-plus/src/size.ts:56]: Check this

File: packages/cdk8s-plus/src/size.ts [hunk 7]
```diff
@@ -140,9 +138,6 @@ class StorageUnit {
 function convert(amount: number, fromUnit: StorageUnit, toUnit: StorageUnit, options: SizeConversionOptions = {}) {
   const rounding = options.rounding ?? SizeRoundingBehavior.FAIL;
   if (fromUnit.inKibiBytes === toUnit.inKibiBytes) { return amount; }
-  if (Token.isUnresolved(amount)) {
```
Comment (iliapolo) [inline packages/cdk8s-plus/src/size.ts:143]: Check this

File: packages/cdk8s-plus/test/duration.test.ts [hunk 8]
```diff
@@ -90,7 +89,6 @@ test('parse', () => {
 
 test('to human string', () => {
   expect(Duration.minutes(0).toHumanString()).toBe('0 minutes');
-  expect(Duration.minutes(Lazy.numberValue({ produce: () => 5 })).toHumanString()).toBe('<token> minutes');
```
Comment (iliapolo) [inline packages/cdk8s-plus/test/duration.test.ts:93]: Check this

File: packages/cdk8s/test/__snapshots__/chart.test.ts.snap [hunk 9]
```diff
@@ -71,9 +71,7 @@ Array [
     },
     "spec": Object {
       "foo": 123,
-      "implicitToken": Object {
```
Comment (iliapolo) [inline packages/cdk8s/test/__snapshots__/chart.test.ts.snap:74]: Check this

File: packages/cdk8s/test/tokens.test.ts [hunk 10]
```diff
@@ -1,33 +1,19 @@
...
-import { Testing } from '../src';
+import { Lazy } from '../src';
+import { resolve } from '../src/lazy';
 
 test('lazy', () => {
-  // GIVEN
-  const app = Testing.app();
+
   const hello = {
-    number: Lazy.numberValue({ produce: () => 1234 }),
```
Comment (iliapolo) [inline packages/cdk8s/test/tokens.test.ts:9]: Check this whole test

**Review by eladb — DISMISSED** [review]

File: packages/cdk8s-plus/src/size.ts [hunk 11]
```diff
@@ -1,10 +1,8 @@
-import { Token } from 'constructs';
```
Comment (iliapolo) [inline packages/cdk8s-plus/src/size.ts:1]: For simplicity, I just dropped the entire token handling in the `Size` class. Do we really need it?

File: packages/cdk8s-plus/src/duration.ts [hunk 12]
```diff
@@ -1,12 +1,9 @@
-import { Token } from 'constructs';
```
Comment (iliapolo) [inline packages/cdk8s-plus/src/duration.ts:1]: For simplicity, I just dropped the entire token handling in the `Duration` class. Do we really need it?

File: packages/cdk8s-plus/src/size.ts (hunk 11 above)
Comment (eladb) [reply packages/cdk8s-plus/src/size.ts:1]: Okay

File: packages/cdk8s-plus/src/duration.ts (hunk 12 above)
Comment (eladb) [reply packages/cdk8s-plus/src/duration.ts:1]: Okay

**Review by eladb — APPROVED** [review]

**Review by eladb — APPROVED** [review]

**Pull request merged**